psd-web/spec/models/user_spec.rb

```diff
@@ -417,4 +419,81 @@
       end
     end
   end
+
+  describe "devise gems echosystem" do
+    describe "#send_two_factor_authentication_code", :with_test_queue_adpater do
+      subject(:user) { create(:user) }
+
+      # rubocop:disable RSpec/MultipleExpectations, RSpec/ExampleLength
+      it "enqueues a SendTwoFactorAuthenticationJob" do
+        ActiveJob::Base.queue_adapter = :test
+        expect { user.send_new_otp }
+          .to enqueue_job(SendTwoFactorAuthenticationJob)
+                .at(:no_wait)
+                .on_queue("psd")
+                .with do |user_param, code_param|
+          expect(user_param).to be user
+          expect(user.direct_otp).to eq(code_param)
+        end
+      end
+      # rubocop:enable RSpec/MultipleExpectations, RSpec/ExampleLength
+    end
```

Comment:
```suggestion
      it "enqueues a SendTwoFactorAuthenticationJob", :with_test_queue_adapter do
        expect { user.send_new_otp }
          .to enqueue_job(SendTwoFactorAuthenticationJob)
                .at(:no_wait)
                .on_queue("psd")
                .with(user, code_param)
      end
    end
```

A couple of issues here:
- Reset the queue adapter after the example. otherwise all subsequent tests will be using this different `test` adapter. Ideally create a shared example (`with_test_queue_adapter`) which has before/after hooks.
- You should be able to use the `with` style above to avoid having to disable the `RSpec/MultipleExpectations` cop.

Reply:
Seems using `with(user, user.direct_otp)` does not refresh the `user.direct_otp` before defining the expectation.
So gets the expectation with, eg: "12345", then the `user.send_new_otp` refreshes the code for user, eg: "54321", but the expectation was already defined. So when checks the assertion compares "12345" (expected) with "54321" (received) and fails.